docs/proposals/prop-1-proposals.md

```diff
@@ -0,0 +1,153 @@
+the Proposal is actually a bad idea, or has accepted that a competing proposal is a better alternative.
+
+When a Proposal is `Accepted`, `Rejected`, or `Withdrawn`, the status should be updated accordingly. In addition to
+updating the status field, the Resolution header should be added with a link to the relevant PR or thread in the mailing
+list archives.
+
+Proposals can also be superseded by a different Proposal, rendering the original obsolete. The status should be updated
+to `Replaced` and the `Replaced-By` and `Replaces` headers should be added to the original and new Proposals 
+respectively.
+
+Process proposals may have a status of `Active` if they are never meant to be completed, e.g. this proposal.
+
+## Technical Specifications
+
+### Document
+
+Proposal documents live in the `docs/proposals` folder of the `anonlink` library. Proposals are written
+in Markdown (or RST?)
+
+Proposal documents will be named `prop-{n}-{short-name}.md`.
+ 
+This meta proposal is therefore found in the file: `prop-1-proposals.md`.
+
+### Auxiliary Files
+
+Additional files (such as diagrams or scripts) should be placed in a folder with the name `prop-n`.
+
+### Proposal Contents
+
+A proposals should contain a single key feature or new idea. Small enhancements or patches don't need a proposal.
+
+At a minimum a proposal needs to have:
+
+- Abstract - a short (~200 word) abstract.
+- Preamble (Title, List of authors, Proposal Number)
+- Proposal Details
+
+The proposal details will usually include:  
+
+- motivation - Clearly explain why the proposal is being made. 
+- rational - why particular design decisions were made. It should describe alternate designs that were considered and related work
+- technical specification - This should be detailed enough to write an implementation from.
+- rejected ideas - Often various ideas will be proposed that are rejected for some reason. Ideally these ideas should be 
+  recorded along with the reasoning as to why they were rejected.
+- Open Issues
+- Backwards Compatibility
+- references
+
+### Preamble
+
+Each proposal must begin with an [RFC 822](https://tools.ietf.org/html/rfc822.html) style header preamble. The headers
+must appear in the following order. Headers marked with "*" are optional and are described below.
+
+```
+Proposal: <proposal number>
+Title: <proposal title>
+Author: <list of authors real names>
+Status: <Draft | Active | Accepted | Provisional | Deferred | Rejected |
+         Withdrawn | Final | Superseded>
+Created: <date first written, in yyyy-mm-dd format>
+* Replaces: <Proposal number>
+* Replaced By: <Proposal number>
+```
+
+All proposals should begin with a status of `Draft`.
+
+## Open Issues
+
+- Should proposals just apply to the anonlink library, or anything related to the project.
+- The process for accepting/rejecting/adopting a proposal
+- Proposals are discussed on the anonlink mailing list and/or GitHub PR?
```

Comment:
We currently don't have a mailing list...

Reply:
Erm yes we do - https://groups.google.com/forum/#!forum/anonlink

See the [readme](https://github.com/data61/anonlink/#discussion)